NBitcoin/Protocol/Node.cs

```diff
@@ -680,43 +691,51 @@ internal Node(NetworkAddress peer, Network network, NodeConnectionParameters par
 			_Connection = new NodeConnection(this, socket);
 			socket.ReceiveBufferSize = parameters.ReceiveBufferSize;
 			socket.SendBufferSize = parameters.SendBufferSize;
-			using(TraceCorrelation.Open())
+
+			using (TraceCorrelation.Open())
 			{
 				try
 				{
-					var completed = new ManualResetEvent(false);
-					var args = new SocketAsyncEventArgs();
-					args.RemoteEndPoint = peer.Endpoint;
-					args.Completed += (s, a) =>
+					using (var completedEvent = new ManualResetEvent(false))
 					{
-						Utils.SafeSet(completed);
-					};
-					if(!socket.ConnectAsync(args))
-						completed.Set();
-					WaitHandle.WaitAny(new WaitHandle[] { completed, parameters.ConnectCancellation.WaitHandle });
-					parameters.ConnectCancellation.ThrowIfCancellationRequested();
-					if(args.SocketError != SocketError.Success)
-						throw new SocketException((int)args.SocketError);
-					var remoteEndpoint = (IPEndPoint)(socket.RemoteEndPoint ?? args.RemoteEndPoint);
-					_RemoteSocketAddress = remoteEndpoint.Address;
-					_RemoteSocketEndpoint = remoteEndpoint;
-					_RemoteSocketPort = remoteEndpoint.Port;
-					State = NodeState.Connected;
-					ConnectedAt = DateTimeOffset.UtcNow;
-					NodeServerTrace.Information("Outbound connection successfull");
-					if(addrman != null)
-						addrman.Attempt(Peer);
+						using (var socketEventManager = NodeSocketEventManager.Create(completedEvent, peer.Endpoint))
+						{
+							//If the socket connected straight away (synchronously) unblock all threads.
+							if (!socket.ConnectAsync(socketEventManager.SocketEvent))
+								completedEvent.Set();
+
+							//Otherwise wait for the socket connection to complete OR if the operation got cancelled.
+							WaitHandle.WaitAny(new WaitHandle[] { completedEvent, parameters.ConnectCancellation.WaitHandle });
+
+							parameters.ConnectCancellation.ThrowIfCancellationRequested();
+
+							if (socketEventManager.SocketEvent.SocketError != SocketError.Success)
+								throw new SocketException((int)socketEventManager.SocketEvent.SocketError);
+
+							var remoteEndpoint = (IPEndPoint)(socket.RemoteEndPoint ?? socketEventManager.SocketEvent.RemoteEndPoint);
+							_RemoteSocketAddress = remoteEndpoint.Address;
+							_RemoteSocketEndpoint = remoteEndpoint;
+							_RemoteSocketPort = remoteEndpoint.Port;
+
+							State = NodeState.Connected;
+

```

Comment:
`ConnectedAt = DateTimeOffset.UtcNow;` seems to have disappeared. Is it intentional?

Reply:
Ah yes, I saw that this property isn't actually being used anywhere? I checked FullNode as well. Is this correct?

Reply:
Maybe it's not used but at the moment it's populated.
If someone does node.ConnectedAt they'll have a date, but if you remove it, it'll be null which would be surprising for them.

Reply:
Should we maybe just remove this property all together?